batch/batch/driver/instance_pool.py

```diff
@@ -129,6 +70,54 @@ def __init__(self, app, machine_name_prefix):
 
         self.name_instance = {}
 
+    async def update_zones(self):
+        northamerica_regions = {
+            # 'northamerica-northeast1',
+            'us-central1',
+            'us-east1',
+            'us-east4',
+            'us-west1',
+            'us-west2',
+            'us-west3',
+            'us-west4'
+        }
+
+        new_zones = []
+        new_zone_weights = []
+
+        async for r in await self.compute_client.list('/regions'):
+            name = r['name']
+            if name not in northamerica_regions:
+                continue
+
+            quota_remaining = {
+                q['metric']: q['limit'] - q['usage']
+                for q in r['quotas']
+            }
+
+            remaining = quota_remaining['PREEMPTIBLE_CPUS'] // self.worker_cores
+            if self.worker_local_ssd_data_disk:
+                remaining = min(remaining, quota_remaining['LOCAL_SSD_TOTAL_GB'] // 350)
```

Comment:
This should be 375 instead of 350. The 350 number comes from the amount of space we reserve for jobs (25 GB goes for Docker images etc.)